### PR TITLE
Postgres batching refinements

### PR DESCRIPTION
**Pull request by nmarton-da**
* Switch from shortcut to support nullable types to explicit Optional PGField-s
* Add scaladoc to type parameters of PGField
* Typo fix

changelog_begin
changelog_end

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/participant-integration-api/src/main/scala/platform/store/backend/postgresql/PGField.scala [hunk 1]
```diff
@@ -9,10 +9,13 @@ import java.time.format.DateTimeFormatter
 
 import scala.reflect.ClassTag
 
-/** Type parameters TO and CONVERTED need to comply to the following contract:
-  * - either both of them nullable (AnyRef)
-  * - or neither of them nullable (CONVERTED AnyVal, TO AnyVal or AnyRef which will be never null)
-  * This is important, since absence is automatically signaled with null references (and we still would want to support primitives)
+/** @tparam FROM is an arbitrary type which from we can extract the data of interest for the particular column
```
Comment (tudor-da) [inline ledger/participant-integration-api/src/main/scala/platform/store/backend/postgresql/PGField.scala:12]:
```suggestion
/** @tparam FROM is an arbitrary type from which we can extract the data of interest for the particular column
```

File: ledger/participant-integration-api/src/main/scala/platform/store/backend/postgresql/PGField.scala [hunk 2]
```diff
@@ -70,9 +81,23 @@ private[postgresql] final case class PGStringArray[FROM](extract: FROM => Iterab
...
+
 private[postgresql] final case class PGBytea[FROM](extract: FROM => Array[Byte])
     extends TrivialPGField[FROM, Array[Byte]]
 
+private[postgresql] final case class PGByteaOptional[FROM](extract: FROM => Option[Array[Byte]])
+    extends TrivialOptionalPGField[FROM, Array[Byte]]
+
 private[postgresql] final case class PGIntOptional[FROM](extract: FROM => Option[Int])
     extends PGField[FROM, Option[Int], java.lang.Integer] {
   override def convert: Option[Int] => Integer = _.map(x => x: java.lang.Integer).orNull
```
Comment (tudor-da) [inline ledger/participant-integration-api/src/main/scala/platform/store/backend/postgresql/PGField.scala:103]:
```suggestion
  override def convert: Option[Int] => Integer = _.map(Int.box).orNull
```

Comment (tudor-da) [reply ledger/participant-integration-api/src/main/scala/platform/store/backend/postgresql/PGField.scala:103]: Also 10 lines lower

**Review by rautenrieth-da — APPROVED** [review]

**Review by tudor-da — APPROVED** [review]
Nice refinement

File: ledger/participant-integration-api/src/main/scala/platform/store/backend/postgresql/PGField.scala (hunk 2 above)
Comment (nmarton-da) [reply ledger/participant-integration-api/src/main/scala/platform/store/backend/postgresql/PGField.scala:103]: groovy!

**Pull request merged**